### ISSUE TITLE
NFT purchase view changes

### ISSUE THREAD

**Issue by toiletgranny** [Joystream/atlas#2224]
**AC:**

- [x] Fix issues 3,4,5 pointed out in Tomek's comment below
- [x] Update the section with top bid and your bid
- [x] Add change price flow
- [x] Make sure in open auction your new bid doesn't have to be higher than the top one

---

This issue gathers all my recommendations for things we need to address in [NFT purchase view designs](https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-purchase-view?node-id=0%3A1) due to recent findings and changes to our understanding of NFTs in Joystream.

## Changing bid
With the recent design NFT changes and adjustments, it has become apparent that each member can have only _one_ bid placed under a particular NFT at a time.

For this reason, if I, as a user, have placed a bid in an open auction, and that bid got outbid later by someone else, to make a new offer I can either withdraw my last bid and place a new one, or change the one I placed previously. And the latter is a flow that we're currently missing in the designs for the NFT purchase view. 

In other words, designs for the NFT purchase view need to be updated so that the below user story is supported:
_**As a content viewer, I want to be able to change my bid in an open auction.**_

Please note that while changing my bid in an open auction I can change it to be either higher or lower, as long as it matches the starting price (min bid).

## Starting price & "Current bid"
1. In open auctions, unlike with english auctions, there's no concept of a "current bid". Any bid can win in an open auction, it doesn't need to be the most recent or the highest one. For this reason, perhaps we can rephrase this to say "Top bid"? I believe that this way we can have this working both for english and open auctions, without needing to introduce a logic that would change this label depending on the auction type. An additional bonus would be preserving consistency with the rest of the designs, as we're already using the term "Top bid" in a few places.
2. There's no concept of a minimum bid in open auctions in a way it exists in english auctions. The only requirement a bid needs to meet in an open auction is being equal or higher than the _starting price_, right? Perhaps we might want to consider changing the wording here for open auctions to say "starting price", rather than saying "minimum bid"? I don't think this is necessary, so I'll leave this one up to you to decide, as an author of the initial designs for this view. 

![CleanShot 2022-03-08 at 12 12 46@2x](https://user-images.githubusercontent.com/35307309/157227582-0b74432b-7b20-42db-bf63-ec861cb19315.png)

cc @kdembler @dmtrjsg

**Comment by kdembler** [comment]
> There's no concept of a minimum bid in open auctions in a way it exists in english auctions

Not sure what exactly you mean by that, but the concept is pretty much the same for open and English auction. I'm gonna use term "starting price", but this can be "minimum bid" if we desire, it means the same thing. Every auction will have this starting price and any bid to any auction always has to be bigger that this value. In English auction, the bid also has to bigger than the current top bid.

**Comment by TCzechowski** [comment]
> With the recent design NFT changes and adjustments, it has become apparent that each member can have only one bid placed under a particular NFT at a time.

1. Do we have any information if it is just on Joystream or is it common on other platforms? I'll do a quick research but maybe you know already :)

> Perhaps we might want to consider changing the wording here for open auctions to say "starting price", rather than saying "minimum bid"?

2. I'm with @kdembler on this one. Even if it is a "starting price", "reservation price" (seen that on other platforms), or however we want to name it, it is still a **minimum bid** I have to place, so for me "minimum bid" is universal and means the same thing, and I don't think we need to add complexity to it :) 

3. Top bid sounds good. I'll change that later today!

**Comment by toiletgranny** [comment]
@kdembler @TCzechowski You guys are correct, I overcomplicated things with the minimum bid/starting price. Keeping the "minimum bid" sounds good!

> Do we have any information if it is just on Joystream or is it common on other platforms? I'll do a quick research but maybe you know already :)

I've seen it also being the case on Rarible. Either way, I'm not sure if it's anything we can change right now, so if that's the reason you're doing the research, it's probably not necessary at this point.

**Comment by TCzechowski** [comment]
> I've seen it also being the case on Rarible. Either way, I'm not sure if it's anything we can change right now, so if that's the reason you're doing the research, it's probably not necessary at this point.

Nope, just wanted to see how other platforms handle "changing" the bid, and asked to know which ones to check first

**Comment by TCzechowski** [comment]
Hi @kdembler @rafalpawlow, I've reviewed the implementation and also made some improvements to support the "Changing bid" flow, you can find the summary below:

## Implementation feedback

**1. WIthdrawal information**
I haven't specified that well so I just want to make sure - is this {X} going to be dynamic, but for now is just mocked because we have no data? 
![image](https://user-images.githubusercontent.com/22870059/158191368-54147a14-24c6-4473-89c4-5c97b433fbf8.png)

**2. Blocks tooltip**
Same question here - The ending block is going to be different for each auction, correct? :) 
![image](https://user-images.githubusercontent.com/22870059/158191902-02d628c4-630f-473a-aee0-001125013f61.png)

**3. Payment split**
The Joystream logo is too big. It is now 32px tall, but it should be 24px
![image](https://user-images.githubusercontent.com/22870059/158192472-bfc21203-7a1e-4579-a32a-2fa638f42d3d.png)

**4. Mobile view**
When we open up the overlay on mobile, it is scrolled to the bottom of the page. Can we display it starting from the top?

**5. Cropping bidding form**
That's probably my fault in the documentation, but when I scroll down the page, the right column gets cropped at the same height as the NFT card on the left ends. Can we make it so it gets cropped when it reaches the top navigation bar (I hope it makes sense 😅 )? 
How it is:
![image](https://user-images.githubusercontent.com/22870059/158194835-627cdf2e-8de0-4d5f-85b7-70da40e3dd4c.png)

How it should be:
![image](https://user-images.githubusercontent.com/22870059/158194739-6bed033e-0415-404a-9684-534680de8cb1.png)

## Visual updates & Changing bid view

**1. Price field pill**
Can we please change the pill variant from Overlay to Default? 🙏 
![image](https://user-images.githubusercontent.com/22870059/158196069-9f42f6ef-2428-4d03-be05-31e47598b167.png)

**2. Auction timer**
Let's get rid of the divider (use "/" instead), and unify all of the fonts to be H200
![image](https://user-images.githubusercontent.com/22870059/158197810-81cf319f-bf34-4009-b172-1f51f0030694.png)

**3. Top bid component**
We've had to reorganize the Current bid (now called "Top bid") component to support changing bid flow
![image](https://user-images.githubusercontent.com/22870059/158199183-4d6b1ccc-58e1-4bd0-8a3f-ae45b240ee46.png)

Figma:
https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-purchase-view?node-id=1035%3A108580

Change bid flow:
https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-purchase-view?node-id=1579%3A25258


I hope I didn't add a lot of effort for you guys, thanks!

**Comment by kdembler** [comment]
**In response to the implementation review:**
Points 1 and 2 will be done as part of separate task that hooks this view up with real data: #1860
What you mentioned in 3, 4 and 5 needs to be fixed, thanks!

---

**In response to new changes:**
I have some feedback for what you mentioned in point 3:
1. I see we're getting rid of the member avatar when displaying who has the top bid at the moment. Any specific reason for that? I like it there, I think we could also stay consistent with how we display it in the NFT Widget, as seen [here](https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2077%3A143765). What do you think?
2. What's up with the icon we're now using for the token? I think it's the monochrome variant but with different color applied to it, what's the reason behind this? I'm a bit concerned with the number of different variants we use for this icon, shouldn't we be more consistent?

**Comment by TCzechowski** [comment]
@kdembler 
1. We tried various combinations of this component to support bid changing flow. In this case, having a member avatar didn't look good as having two member avatars, two tokens made a lot of visual noise in this area (for Top bid and Your bid). We also wanted to save as much vertical space as possible, and if we'd want to keep the avatar there, we'd have to increase paddings/spacings, so it would be counter to our intended goal. That's why we decided to remove that from this view, as it is not that important on this view anyway :)
2. Long story short - silver variant (the one with gradients) had some implementation issues and didn't look that good after implementing it there, so we've decided to swap it with the Monochrome variant **BUT**, the default color of monochrome (text/strong) was well, too strong and too prominent. That's why we decided to override the color to text/default.

Hope that makes sense!

**Comment by kdembler** [comment]
1. I don't understand. I'm not talking about current user's avatar, but avatar of the top bidder. I've linked the widget design which handles this case without using any additional vertical space. I think everywhere in the app we show the member's handle along with their avatar, I think we should do the same here.
2. What implementation issues are you referring to here? I'm just concerned with styling this on one-off basis, instead of using a defined variant. This way potentially we risk things getting out of sync in the future, but let's proceed I guess

**Comment by TCzechowski** [comment]
1. Just to clarify: You mean something like that in the picture below (right side, left side is the current version), right? If so, then I guess we could do that, but it adds a bit more noise and I don't think it adds that much value for the user. I understand the consistency point, though. I'm a bit torn here, WDYT @toiletgranny? Any preferences on your side? It would also require adding 16px variant to the Avatar component
![image](https://user-images.githubusercontent.com/22870059/158811242-476196de-6b59-4deb-b29b-baad8621bf34.png)

2. I'm referring to those highlights that can be seen in Figma (left side) and not in the implemented version (right side). "Implementation issue" might have been a wrong word here, but I hope the screenshot below cleared things up. Maybe we should add another, gray variant to this component instead of overriding the colors here, @toiletgranny? 
![image](https://user-images.githubusercontent.com/22870059/158808863-42bf14d2-08b6-494f-b42f-3701c12c7e5d.png)

**Comment by kdembler** [comment]
1. Yep, exactly what I meant. I think it does add value since identifying someone via avatar is much quicker and we always display member handle along with the avatar.
2. I'll leave it up to you

**Comment by toiletgranny** [comment]
+1 to adding avatars next to bids. 

> It would also require adding 16px variant to the Avatar component

I understand why you want to add a `16px` variant to the Avatar component, it indeed looks slicker than pairing a `24px` avatar with a `16px` icon. But there are other variants of this component (with a button or info inside) that would need to be handled in this extremely small size, so I'm afraid it's not as easy as it might seem. Ultimately, I would vote for using a `24px` avatar, just like in the NFT widget component that I and @kdembler have suggested, and we can revisit this once we refactor the Avatar component as a whole in the near future.

**Comment by toiletgranny** [comment]
Regarding the color of the JOY token icon, I spoke to @kdembler to clarify that the purpose of having a "Monochrome" variant is to be able to change the color of the icon freely, just like with any other icon. How is this handled on the implementation side, I think it's best to leave this up for the tech team to decide.
